hibernate-core/src/test/java/org/hibernate/orm/test/mapping/inheritance/discriminator/AccountOwner.hbm.xml

```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0"?>
+<!--
+  ~ Hibernate, Relational Persistence for Idiomatic Java
+  ~
+  ~ License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+  ~ See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+  -->
+<!DOCTYPE hibernate-mapping PUBLIC
+        "-//Hibernate/Hibernate Mapping DTD 3.0//EN"
+        "http://www.hibernate.org/dtd/hibernate-mapping-3.0.dtd">
+
+<hibernate-mapping
+        package="org.hibernate.orm.test.mapping.inheritance.discriminator">
+
+    <class name="AccountOwner" table="AccOwner">
+        <id name="id">
+            <generator class="native"/>
+        </id>
+        <property name="description"/>
+
+        <set name="creditAccounts"
+             cascade="persist">
+            <key column="ownerId"/>
+            <one-to-many class="CreditAccount"/>
+        </set>
+
+        <set name="debitAccounts"
+             cascade="persist">
+            <key column="ownerId"/>
+            <one-to-many class="DebitAccount"/>
+        </set>
+    </class>
+
+    <class name="AbstractAccount" table="Accounts" abstract="true" dynamic-update="true">
+        <id name="id">
+            <generator class="native"/>
+        </id>
+
+        <discriminator column="DISC" force="true" insert="true" type="java.lang.String" not-null="true"/>
```

Comment:
The hbm version works, because you are using `force="true"`, which is the workaround I mentioned to you

```suggestion
        <discriminator column="DISC" insert="true" type="java.lang.String" not-null="true"/>
```

Reply:
True, I forgot to remove that. I should add though, that it *does* actually work without the force.